.github/workflows/agent-packaging.yml

```diff
@@ -0,0 +1,104 @@
+name: Release Agent v3
+
+on:
+  workflow_call:
+    inputs:
+      packageVersion:
+        type: string
+        default: "3.0.0"
+      packageBuildNo:
+        type: string
+        default: "1"
+      tagRelease:
+        type: boolean
+        default: false
+      releaseBranch:
+        type: string
+        required: true
+      signingKeyFile:
+        type: string
+        required: true
+      uploadUrl:
+        type: string
+        default: "https://up-ap.nginx.com"
+    secrets:
+      INDIGO_GPG_AGENT:
+
+
+env:
+  NFPM_VERSION: 'v2.35.3'
+  GOPROXY: "https://${{ secrets.ARTIFACTORY_USER }}:${{ secrets.ARTIFACTORY_TOKEN }}@azr.artifactory.f5net.com/artifactory/api/go/f5-nginx-go-dev"
+
+permissions:
+  contents: read
+  id-token: write
+
+defaults:
+  run:
+    shell: 'bash -Eeo pipefail -x {0}'
+
+jobs:
+  build-and-upload-packages:
+    name: Build and upload Agent packages
+    runs-on: ubuntu-24.04-amd64
+    permissions:
```

Comment:
Do we need to duplicate permissions here if we already have it defined on the top-most level?